app/assets/javascripts/training/components/modals/transfer_step3.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import SelectableBox from '../../../components/common/selectable_box.jsx';
+import { transferModules } from '../../../actions/training_modification_actions.js';
+import { extractCategoriesFromHtml } from '../../../utils/training_utils.js';
+
+// Choose category to which user wants to transfer module
+const TransferStep3 = ({ transferInfo, setTransferInfo, step, setStep, setSubmitting }) => {
+  const { library_id } = useParams();
+  const [categories, setCategories] = useState([]);
+  const dispatch = useDispatch();
+
+  const handleCategorySelection = (selectedCategory) => {
+    setTransferInfo(prev => ({ ...prev, destinationCategory: selectedCategory }));
+  };
+
+  const submitHandler = () => {
+    setSubmitting(true);
+    dispatch(transferModules(library_id, transferInfo, setSubmitting));
+  };
+
+  useEffect(() => {
+    const setCategoriesFromHtml = () => {
+      const extractedCategories = extractCategoriesFromHtml();
+      setCategories(extractedCategories.filter(category => category.name !== transferInfo?.sourceCategory));
+      const event = new CustomEvent('categoriesReady', { detail: { categories: extractedCategories } });
+      document.dispatchEvent(event);
+    };
+
+    if (document.readyState === 'complete' || document.readyState === 'interactive') {
+      setCategoriesFromHtml();
+    } else {
+      document.addEventListener('DOMContentLoaded', setCategoriesFromHtml);
+    }
+
+    return () => {
+      document.removeEventListener('DOMContentLoaded', setCategoriesFromHtml);
+    };
+  }, [transferInfo.sourceCategory]);
```

Comment:
Having a hard time understanding the reasoning behind this. Why's do we need a custom event — and won't the categories be always present in the DOM, since its server rendered? I'm sure I'm missing something

Reply:
Initially, I used the custom event to ensure the categories were fully extracted and ready for use, providing a clear separation between extraction and component logic. Based on your feedback, I realized this was unnecessary since the categories are server-rendered and always present in the DOM. So, now I have simplified the code by directly setting the categories in the state without the custom event.

Reply:
Yup, this looks much better now